image/verify/src/verifier.rs

```diff
@@ -775,12 +775,17 @@ impl<Env: ImageVerificationEnv> ImageVerifier<Env> {
             caliptra_cfi_lib::cfi_assert_eq_12_words(&verify_info.digest, &actual);
         }
 
-        // Overflow/underflow is checked in verify_toc
+        // Validate ICCM bounds with overflow protection
```

Comment:
are these necessary? The previous comment says that these are checked in verify_toc. Is that not the case?